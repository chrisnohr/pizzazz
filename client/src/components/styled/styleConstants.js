// colours
export const textColor = '#6C757D';
export const fadedText = '#AAB0B5';
export const borderColor = '#CED4DA';
export const white = '#FFF';
export const beige = '#E7DDCF';
export const pink = '#FFC2C0';
export const green = '#C4F2CB';
export const yellow = '#F7E9B7';
export const letterColor = '#555';
export const blue = '#BDEAFE';
// spacing
export const spacer = '3rem';
export const padding = '0.8rem 1.6rem';
export const respPad = '3vw 6vw';
export const respSpacer = '9vw';
// shadow
export const shadow = '1.8px 2.4px 1.8px rgba(0, 0, 0, 0.05)';
// text
export const respText = '4vw';
export const textSize = '1.1rem';
// border
export const border = `.5px solid ${borderColor}`;
// helpers
export const bottomSpacer = `
  margin-bottom: ${respSpacer};
  @media screen and (min-width: 501px) {
    margin-bottom: ${spacer};
  }
`;
export const centerH = `
  margin-left: auto;
  margin-right: auto;
`;
