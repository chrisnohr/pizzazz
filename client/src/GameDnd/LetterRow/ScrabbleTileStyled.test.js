import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import { LetterStyled } from './ScrabbleTileStyled';
import { green, yellow, letterColor } from '../../styleConstants';

describe('<ScrabbleTileStyled />', () => {
  it('renders a dark gray letter when the letter is not 8 or 0', () => {
    const wrapper = renderer.create(<LetterStyled letter="p" isValid={false} />).toJSON();
    expect(wrapper).toHaveStyleRule('color', letterColor);
  });
  it('renders an invisible letter when the letter is 8', () => {
    const wrapper = renderer.create(<LetterStyled letter="8" isValid={false} />).toJSON();
    expect(wrapper).toHaveStyleRule('color', 'rgba(0,0,0,0)');
  });
  it('renders an invisible div when the letter is 0', () => {
    const wrapper = renderer.create(<LetterStyled letter="0" isValid={false} />).toJSON();
    expect(wrapper).toHaveStyleRule('opacity', '0');
  });
  it('renders a yellow background when isValid is false', () => {
    const wrapper = renderer.create(<LetterStyled letter="8" isValid={false} />).toJSON();
    expect(wrapper).toHaveStyleRule('background-color', yellow);
  });
  it('renders a green background when isValid is true', () => {
    const wrapper = renderer.create(<LetterStyled letter="8" isValid={true} />).toJSON();
    expect(wrapper).toHaveStyleRule('background-color', green);
  });

  // snapshot tests
  it('matches the snapshot', () => {
    const wrapper = shallow(<LetterStyled letter="e" isValid={false} />);
    expect(wrapper).toMatchSnapshot();
  });
});
