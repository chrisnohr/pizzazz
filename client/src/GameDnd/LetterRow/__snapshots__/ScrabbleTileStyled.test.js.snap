// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScrabbleTileStyled /> matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForwardRef
    isValid={false}
    letter="e"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "forwardedComponent": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": false,
          "lastClassName": "kCqDVu",
          "rules": Array [
            "
  opacity: ",
            [Function],
            ";
  user-select: none;
  border: 1px solid ",
            "#CED4DA",
            ";
  border-radius: 0.5vw;
  color: ",
            [Function],
            ";
  background-color: ",
            [Function],
            ";
  font-weight: 400;
  font-size: 8vw;
  text-align: center;
  line-height: 12.2vw;
  box-shadow: ",
            "1.8px 2.4px 1.8px rgba(0, 0, 0, 0.05)",
            ";
  @media screen and (min-width: 501px) {
      font-size: 40px;
      line-height: 65px;
      border-radius: 4px;
  }
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
      "forwardedRef": null,
      "isValid": false,
      "letter": "e",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "forwardedComponent": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "lastClassName": "kCqDVu",
            "rules": Array [
              "
  opacity: ",
              [Function],
              ";
  user-select: none;
  border: 1px solid ",
              "#CED4DA",
              ";
  border-radius: 0.5vw;
  color: ",
              [Function],
              ";
  background-color: ",
              [Function],
              ";
  font-weight: 400;
  font-size: 8vw;
  text-align: center;
  line-height: 12.2vw;
  box-shadow: ",
              "1.8px 2.4px 1.8px rgba(0, 0, 0, 0.05)",
              ";
  @media screen and (min-width: 501px) {
      font-size: 40px;
      line-height: 65px;
      border-radius: 4px;
  }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
        "forwardedRef": null,
        "isValid": false,
        "letter": "e",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
